### PR TITLE
API view for syncing users from the bot #yumr

### PR DESCRIPTION
**Pull request by gdude2002**

### CODE REVIEW

File: pysite/views/api/bot/user.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from pysite.constants import ErrorCodes
+from pysite.decorators import valid_api_key
+
+REQUIRED_KEYS = [
+    "user_id",
+    "role"
+]
+
+
+class TagView(APIView, DBViewMixin):
```
Comment (lemonsaurus) [inline pysite/views/api/bot/user.py]: why is this called TagView?

File: pysite/views/api/bot/user.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    @valid_api_key
+    def post(self):
+        data = request.get_json()
+
+        if not isinstance(data, list):
+            data = [data]
+
+        for user in data:
+            if not all(k in user for k in REQUIRED_KEYS):
```
Comment (lemonsaurus) [inline pysite/views/api/bot/user.py:29]: how about we repurpose the `valid_api_key` decorator to be `valid_api_request` instead, then we make `required_keys` (maybe rename to `required_params`?) be a class attribute, and then the decorator can check if they're present.

That way we only have to define the required params and all the validation is handled automatically.

**Review by lemonsaurus — CHANGES_REQUESTED** [review]

File: pysite/views/api/bot/user.py (hunk 2 above)
Comment (gdude2002) [reply pysite/views/api/bot/user.py:29]: My plan is to have API key management via the database, so it doesn't make sense to supply sets of API keys here.

EDIT: I'm stupid, one second..

Comment (gdude2002) [reply pysite/views/api/bot/user.py:29]: I think that might be a little complex - what do we do if a list is expected?

**Review by lemonsaurus — APPROVED** [review]

**Pull request merged**